tests/lib/tools/tests.info

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash -e
+
+show_help() {
+	echo "usage: tests.info <CMD>"
+	echo
+	echo "Supported commands:"
+	echo "    is-snapd-pkg-repo: indicates when the snap packages is from the repository"
+	echo "    is-reexec-enabled: indicates when snapd package is re-executing to snapd snap"
+}
+
+
+is_snapd_pkg_repo() {
+    [ "$SNAPD_DEB_FROM_REPO" = true ] && [ -n "$(command -v apt)" ]
+}
+
+is_reexec_enabled() {
+    snap debug execution snap | grep -q "is-reexecd: true"
```

Comment:
I think this is somewhat misleading. There are two values:

- is snapd re-execution requested
- has snapd actually re-executed

I'd rename this sub-command (function and external command name) to indicate what is actually being measured. One quick test of this is to have both snap and non-snap at exactly the same version. In that case we just don't re-exec unless we _force_ re-exec by setting SNAP_REEXEC=1

Reply:
I updated this, now we have 2 commands to check is re-exec is requested and another to check is re-exec is performed.